Manatee.Json.Tests/XmlExtensionsTest.cs

```diff
@@ -524,6 +524,71 @@ public void RoundTrip_StartingWithXml()
 
 			Assert.AreEqual(expected.ToString(), actual.ToString());
 		}
+		[Test]
+		public void RoundTrip_StartingWithRepeatingXml()
+		{
+			var expected = XElement.Parse(@"<Items><Item>1</Item><Item>2</Item><Item>3</Item></Items>");
+			var expectedJson = JsonValue.Parse(@"{""Items"":{""Item"":[1,2,3]}}");
+
+			var toJson = expected.ToJson();
+			Assert.AreEqual(expectedJson, toJson);
+
+			var actual = toJson.ToXElement(null);
+
+			Assert.AreEqual(expected.ToString(), actual.ToString());
+		}
+		[Test]
+		public void RoundTrip_StartingWithMoreComplexXml()
+		{
+			var expected = XElement.Parse(@"<Items><Item k=""1"" v=""a"" /><Item k=""2"" v=""b"" /><Item k=""3"" v=""c"" /></Items>");
+			var expectedJson = JsonValue.Parse(@"{""Items"":{""Item"":[{""-k"":1,""-v"":""a""},null,[{""-k"":2,""-v"":""b""},null],[{""-k"":3,""-v"":""c""},null]]}}");

```

Comment:
Seems like this is off:
```suggestion
			var expectedJson = JsonValue.Parse(@"{""Items"":{""Item"":[{""-k"":1,""-v"":""a""},null],[{""-k"":2,""-v"":""b""},null],[{""-k"":3,""-v"":""c""},null]]}}");

```

Reply:
I make the assumption at this point that the `toJson` implementation is as intended, so expect what it currently generates, which is an array that contains both the attributes and the sibling elements of the same key.

The suggested change has, for one thing, unbalanced square brackets (3*`[` and 4*`]`).

Reply:
Yeah I missed removing the end one, that's my bad. Seems unexpected that they would be nested like that.

We use some of the ToXElement features, but I haven't confirmed if this would affect anything we do.

Reply:
A result of using an implicit reading of an array, overloading the meanings of the array of attributes (part of XML), and the array of similarly-named siblings (because JavaScript doesn't allow multiple fields with the same name)

The change should only affect cases where an array includes both a collection of attributes (names with the leading "-" sigil) at the start *and* sibling items following; heretofore such arrays would simply throw an `XmlException` processing the decorated names.